src/main/java/com/tenx/ms/retail/order/domain/OrderDetailEntity.java

```diff
@@ -0,0 +1,89 @@
+package com.tenx.ms.retail.order.domain;
+
+import com.tenx.ms.retail.order.util.OrderDetailStatusEnum;
+import com.tenx.ms.retail.product.domain.ProductEntity;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.Id;
+import javax.persistence.Column;
+import javax.persistence.UniqueConstraint;
+import javax.persistence.ManyToOne;
+import javax.persistence.JoinColumn;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Enumerated;
+import javax.persistence.EnumType;
+import javax.persistence.GenerationType;
+
+/**
+ * Created by goropeza on 28/08/16.
+ */
+@Entity
+@Table(name = "order_detail", uniqueConstraints = @UniqueConstraint(name = "unique_order_product_status",
+    columnNames = {"detail_order_id", "detail_product_id", "detail_status"}))
+public class OrderDetailEntity {
+    private Long id;
+    private OrderEntity order;
+    private ProductEntity product;
+    private OrderDetailStatusEnum status;
+    private Long count;
+
+    public OrderDetailEntity() {}
+
+    public OrderDetailEntity(OrderEntity order, ProductEntity product, OrderDetailStatusEnum status, Long count) {
+        this.order = order;
+        this.product = product;
+        this.status = status;
+        this.count = count;
+    }
+
+    @Id
```

Comment:
I find it easier to read if all these annotations are defined in the variable declaration place at the top of the class. This suggestion also applies to other entity classes